cmd/syncthing/connections.go

```diff
@@ -330,10 +334,10 @@ func (s *connectionSvc) connect() {
 				l.Debugln("Trying relay connections to", deviceID, relays)
 			}
 
-			s.lastRelayCheck[deviceID] = time.Now()
-
 			for _, addr := range relays {
-				uri, err := url.Parse(addr)
+				s.lastRelayCheck[deviceID] = time.Now()
```

Comment:
This handles the synthetic delay we had for global discovery at startup before, the five seconds.

Without that delay we do the first connection attempt before we have global discovery up, and thus have no relays. But we still marked the `lastRelayCheck` so we wouldn't retry relays for ten minutes.

With this move we only mark the relays as tried if there were any relays, so we'll actually try the relays as soon as they become available from global discovery, even if it's a minute after initial startup.